rq/job.py

```diff
@@ -393,6 +393,41 @@ def dependents_key(self):
         """The Redis key that is used to store job dependents hash under."""
         return self.dependents_key_for(self.id)
 
+    @property
+    def dependencies_key(self):
+        return '{0}{1}:dependencies'.format(self.redis_job_namespace_prefix, self.id)
+
+    def fetch_dependencies(self, watch=False, pipeline=None):
+        """
+        Fetch all of a job's dependencies. If a pipeline is supplied, and
+        watch is true, the set WATCH on all the keys of all dependencies.
+
+        Returned jobs will NOT use the pipeline supplied.
+        """
+        if pipeline is None:
+            pipeline = pipeline or self.connection
+
+        if watch and self._dependency_ids:
+            pipeline.watch(*self._dependency_ids)
+
+        results = [
+            pipeline.hgetall(self.key_for(job_id))
```

Comment:
Since we need to evaluate results right away, we shouldn't use `pipeline` here, I think we should use `self.connection`.

Also, I think it's about time we have an operation to fetch jobs in bulk. If you have the time, a separate PR implementing `Job.fetch_many(self.connection, [job_id_1, job_id_2])` would be much appreciated. Otherwise, it's also ok :)

Reply:
@selwin it looks as though `Job.fetch_many` already exists :smiley: 

https://github.com/rq/rq/blob/dfc753332489af85ea2f9c9188771f8a2d61e400/rq/job.py#L300

Good thinking re: `self.connection` vs. `pipeline` - passing `self`'s connection to `fetch_many` in dfc753332489af85ea2f9c9188771f8a2d61e400 seems to do the trick.

One other thing - the `fetch_many` method used a default value of `None` for connection, but  will always raise an `AttributeError` of `NoneType` is _something_ is not passed. I've removed this default value in d366b788ecef38709507b4730cc516eab437adfb because it seems more correct to enforce the necessity of `connection` in the method's signature.